File: lab05/lab05example01/src/lab05example01_node.cpp
#include "ros/ros.h"
#include <tf2_eigen/tf2_eigen.h>
#include <sensor_msgs/JointState.h>
#include "kdl_kine/kdl_kine_solver.h"
#include <Eigen/Dense>

KDL::JntArray joint_values;

int main(int argc, char **argv)
{

    ros::init(argc, argv, "robotis_jacob_node");
    ros::NodeHandle nh;

    robot_kinematic h_kine;
    ros::Rate loop_rate(10);

    //Initiate the base link (where the kinematic chain starts) and the end-effector link (where it ends). Please look at the
    // urdf file of the respective robot arm for the names.
    std::string base_link_name = "link1";
    std::string ee_link_name = "end_link";

    h_kine.init(base_link_name, ee_link_name);

    while (ros::ok())
    {
        std::cout << "The current pose: " << std::endl;
        std::cout << h_kine.KDLfkine(h_kine.current_joint_position) << std::endl;
        std::cout << "The current Jacobian: " << std::endl;
        std::cout << h_kine.KDLjacob(h_kine.current_joint_position).data << std::endl;

        ros::spinOnce();
        loop_rate.sleep();
    }



    return 0;
}

File: lab05/lab05example01/scripts/lab05example01_node.py
#!/usr/bin/env python

import rospy
from sensor_msgs.msg import JointState
from kdl_kine.kdl_kine_solver import  robot_kinematic



def show_jacobian():

    rospy.init_node('robotis_jacob_node')

    rate = rospy.Rate(10)
    # Initiate the base link(where the kinematic chain starts) and the end-effector link(where it ends).
    # Please look at the urdf file of the respective robot arm for the names.
    h_kine = robot_kinematic('link1', 'end_link')


    while not rospy.is_shutdown():
        print('The current robot pose: ')
        print(h_kine.forward_kinematics(h_kine.current_joint_position))
        print('Jacobian: ')
        print(h_kine.get_jacobian(h_kine.current_joint_position))
        rate.sleep()



if __name__ == '__main__':
    show_jacobian()
